src/app/services/settings/navigation-auto.service.ts

```diff
@@ -1,12 +1,103 @@
 let navigationAutoServiceIsInstantiated: boolean;
 
 class NavigationAutoService {
+	currentFocusElt: any;
+	currentIndex: number;
+	handler: any;
+	timer: ReturnType<typeof setInterval> | null = null;
 
 	constructor() {
 		if (navigationAutoServiceIsInstantiated) {
 			throw new Error('NavigationAutoService is already instantiated.');
 		}
 
 		navigationAutoServiceIsInstantiated = true;
+
+		this.handler = this.createHandler();
+	}
+
+	setNavigationAuto = (value: string): void => {
+		this.resetAutoFocus();
+		if (value !== DEFAULT_VALUE) {
+			this.initializeFocusTracking();
+			let delay = Number(value.split('_')[1]) * 1000;
+			this.setAutoFocus(delay);
+		}
+	}
+
+	setAutoFocus = (delay: number): void => {
+		this.setIntervalFocus(delay);
+	}
+
+	resetAutoFocus = (): void => {
+		window.removeEventListener('focus', this.handler);
+		this.clearInterval();
```

Comment:
Il faudrait déterminer quel intervalle tu nettoies, ici : probablement `this.timer`, non ?